lib/routes/password.js

```diff
@@ -143,56 +145,155 @@ module.exports = function (
         var authPW = Buffer(request.payload.authPW, 'hex')
         var wrapKb = Buffer(request.payload.wrapKb, 'hex')
         var authSalt = crypto.randomBytes(32)
+        var sessionTokenId = request.payload.sessionToken
         var password = new Password(authPW, authSalt, verifierVersion)
-        db.deletePasswordChangeToken(passwordChangeToken)
-          .then(
-            function () {
-              return password.verifyHash()
-                .then(
-                  function (verifyHash) {
-                    return password.wrap(wrapKb)
-                      .then(
-                        function (wrapWrapKb) {
-                          return db.resetAccount(
-                            passwordChangeToken,
-                            {
-                              verifyHash: verifyHash,
-                              authSalt: authSalt,
-                              wrapWrapKb: wrapWrapKb,
-                              verifierVersion: password.version
-                            }
-                          )
-                        }
-                      )
+        var wantsKeys = (request.query.keys === 'true')
```

Comment:
nit: it may be worth promoting `wantsKeys` to a shared helper under `lib/routes/utils/`